TST: numpy-dev job failed with numpy.ndarray size changed

<!-- This comments are hidden when you submit the issue,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/master/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/master/CODE_OF_CONDUCT.md . -->

<!-- Please have a search on our GitHub repository to see if a similar
issue has already been posted.
If a similar issue is closed, have a quick look to see if you are satisfied
by the resolution.
If not please go ahead and open an issue! -->

<!-- Please check that the development version still produces the same bug.
You can install development version with
pip install git+https://github.com/astropy/astropy
command. -->

EDIT: Sorry for the noise, just saw #11116 after I posted this.

### Description
<!-- Provide a general description of the bug. -->

The numpy-dev job is now weekly cron (not daily). When it ran yesterday (2020-11-30), it failed with this error:

```
============================= test session starts ==============================
platform linux -- Python 3.7.9, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
cachedir: .tox/py37-test-devdeps/.pytest_cache

Running tests with Astropy version 4.3.dev292+g0ff46cb53.
Running tests in astropy docs.

Date: 2020-11-30T06:26:19

Platform: Linux-5.4.0-1031-azure-x86_64-with-debian-buster-sid

Executable: /home/runner/work/astropy/astropy/.tox/py37-test-devdeps/bin/python

Full Python Version: 
3.7.9 (default, Oct 28 2020, 13:48:38) 
[GCC 7.5.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.21.0.dev0+d0a457f
Scipy: not available
Matplotlib: 3.3.2+1733.g7c813db74
h5py: not available
Pandas: not available
PyERFA: 1.7.1.1

Using Astropy options: remote_data: none.

rootdir: /home/runner/work/astropy/astropy, configfile: setup.cfg
plugins: openfiles-0.5.0, filter-subpackage-0.1.1, xdist-2.1.0, remotedata-0.3.2, doctestplus-0.8.0, arraydiff-0.3, astropy-header-0.1.2, cov-2.10.1, asdf-2.8.0.dev107+g8ccee8b, forked-1.3.0, hypothesis-5.41.4
collected 0 items / 1 error

==================================== ERRORS ====================================
________________________ ERROR collecting test session _________________________
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/pluggy/manager.py:127: in register
    hook._maybe_apply_history(hookimpl)
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/pluggy/hooks.py:333: in _maybe_apply_history
    res = self._hookexec(self, [method], kwargs)
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/pluggy/manager.py:87: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/conftest.py:58: in pytest_configure
    from astropy.utils.iers import conf as iers_conf
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/utils/iers/__init__.py:2: in <module>
    from .iers import *
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/utils/iers/iers.py:23: in <module>
    from astropy.table import QTable, MaskedColumn
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/table/__init__.py:4: in <module>
    from .column import Column, MaskedColumn, StringTruncateWarning, ColumnInfo
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/table/column.py:22: in <module>
    from ._column_mixins import _ColumnGetitemShim, _MaskedColumnGetitemShim
astropy/table/_column_mixins.pyx:46: in init astropy.table._column_mixins
    ???
E   RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 80 from C header, got 88 from PyObject
```

Even though there is a `cachedir: .tox/py37-test-devdeps/.pytest_cache` defined, I don't think we have set up any form of caching in Actions yet. Example log: https://github.com/astropy/astropy/runs/1472240391?check_suite_focus=true

@saimn mentioned [[Numpy-discussion] Changing the size of PyArrayObject_fields (the ndarray c-struct)](https://mail.python.org/pipermail/numpy-discussion/2020-November/081213.html) and numpy/numpy#16938 .

@mhvk , are you able to advise? Thanks!

--- a/setup.cfg
+++ b/setup.cfg
@@ -119,6 +119,7 @@ filterwarnings =
     ignore:unclosed <socket:ResourceWarning
     ignore:unclosed <ssl.SSLSocket:ResourceWarning
     ignore:numpy.ufunc size changed:RuntimeWarning
+    ignore:numpy.ndarray size changed:RuntimeWarning
     ignore:Importing from numpy:DeprecationWarning:scipy
     ignore:Conversion of the second argument:FutureWarning:scipy
     ignore:Using a non-tuple sequence:FutureWarning:scipy

